hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -774,4 +785,23 @@ object DataSourceOptionsHelper {
       override def apply (input: From): To = function (input)
     }
   }
+
+  implicit def convert[T, U](prop: ConfigProperty[T])(implicit converter: T => U): ConfigProperty[U] = {
+    checkState(prop.hasDefaultValue)
```

Comment:
this could implicitly break when add a new config with no default. i see this improves code quality but we should avoid nice-to-have changes in the last min patch before release.

Reply:
If this will break it will break when the class is loaded, meaning that all the tests using class would be broken, which is very easy to diagnose